allofplos/corpus_class.py

```diff
@@ -0,0 +1,267 @@
+from collections import OrderedDict
+import datetime
+import json
+from pathlib import Path
+import os
+import random
+
+from tqdm import tqdm
+
+from . import corpusdir
+
+from .article_class import Article
+from .corpus_helpers import hash_file
+from .gdrive import (get_zip_metadata, download_file_from_google_drive,
+                     unzip_articles, zip_id, local_zip)
+from .transformations import filename_to_doi, doi_to_path
+
+hash_json = 'corpus_hash.json'
+
+
+class Corpus():
+    """A collection of PLOS articles."""
+
+    def __init__(self, directory=corpusdir, plos_network=False, extension='.xml'):
+        """Creation of an article corpus class."""
+        self.directory = directory
+        self.plos_network = plos_network
+        self.extension = extension
+
+    @property
+    def directory(self):
+        """
+        :returns: directory of corpus
+        :rtype: {str}
+        """
+        return self._directory
+
+    @directory.setter
+    def directory(self, d):
+        """
+        Reset memoized info when changing the path to the corpus.
+        """
+        self._directory = d
+
+    def __repr__(self):
+        """Value of a corpus object when you call it directly on the command line.
+
+        Shows the directory location of the corpus
+        :returns: directory
+        :rtype: {str}
+        """
+        out = "Corpus location: {0}\nNumber of files: {1}".format(self.directory, len(self.files))
+        return out
+
+    @property
+    def iter_file_doi(self):
+        """Generator that returns filename, doi tuples for every file in the corpus.
+
+        Used to generate both DOI and file generators for the corpus. 
+        """
+        return ((file_, filename_to_doi(file_))
+                for file_ in os.listdir(self.directory)
```

Comment:
If you have any other files in the directory that do not match the filename_to_doi expected pattern, this may behave strangely. I'd appreciate if you could make some of those checks more explicit (which unfortunately means probably breaking up that pretty generator comprehension into a couple of steps).